### PR TITLE
Update develop with enforced value fixes

### PR DESCRIPTION
**Pull request by blondfrogs**
- Added Bip Logic for the new ENFORCE_VALUE BIP which will enforce reissue assets to have a value of 0 in the txout.nValue - 62c30f9, 905952c, 787d4c0, 9c8fbc0, 8141607, c455cfc, 185a052, 09578af   
- Added new rpccall `decodeblock` - ae65a29
- Updated GUI version checking to use configure.ac variable -  8c1fb71, 627a6f0, 405e1ee

Code Added by developers:
@akshaynexus - 8c1fb71 
@blondfrogs - 62c30f9, 905952c, 787d4c0, 9c8fbc0, 8141607, c455cfc, 185a052, 09578af, 627a6f0, 405e1ee, ae65a29, e889065

Closes #795

### CODE REVIEW

File: src/chainparams.cpp [hunk 1]
```diff
@@ -151,6 +151,11 @@ class CMainParams : public CChainParams {
         consensus.vDeployments[Consensus::DEPLOYMENT_TRANSFER_SCRIPT_SIZE].nTimeout = 1620324000; // UTC: Thu May 06 2021 18:00:00
         consensus.vDeployments[Consensus::DEPLOYMENT_TRANSFER_SCRIPT_SIZE].nOverrideRuleChangeActivationThreshold = 1714; // Approx 85% of 2016
         consensus.vDeployments[Consensus::DEPLOYMENT_TRANSFER_SCRIPT_SIZE].nOverrideMinerConfirmationWindow = 2016;
+        consensus.vDeployments[Consensus::DEPLOYMENT_ENFORCE_VALUE].bit = 9;
+        consensus.vDeployments[Consensus::DEPLOYMENT_ENFORCE_VALUE].nStartTime = 1593453600; // UTC: Mon Jun 29 2020 18:00:00
+        consensus.vDeployments[Consensus::DEPLOYMENT_ENFORCE_VALUE].nTimeout = 1624989600; // UTC: Mon Jun 29 2020 18:00:00
```
Comment (blondfrogs) [inline src/chainparams.cpp:156]: change to 2021 instead of 2020

File: src/chainparams.cpp [hunk 2]
```diff
@@ -302,6 +307,11 @@ class CTestNetParams : public CChainParams {
         consensus.vDeployments[Consensus::DEPLOYMENT_TRANSFER_SCRIPT_SIZE].nTimeout = 1618509600; // UTC: Thu Apr 15 2021 18:00:00
         consensus.vDeployments[Consensus::DEPLOYMENT_TRANSFER_SCRIPT_SIZE].nOverrideRuleChangeActivationThreshold = 1310;
         consensus.vDeployments[Consensus::DEPLOYMENT_TRANSFER_SCRIPT_SIZE].nOverrideMinerConfirmationWindow = 2016;
+        consensus.vDeployments[Consensus::DEPLOYMENT_ENFORCE_VALUE].bit = 9;
+        consensus.vDeployments[Consensus::DEPLOYMENT_ENFORCE_VALUE].nStartTime = 1593453600; // UTC: Mon Jun 29 2020 18:00:00
+        consensus.vDeployments[Consensus::DEPLOYMENT_ENFORCE_VALUE].nTimeout = 1624989600; // UTC: Mon Jun 29 2020 18:00:00
```
Comment (blondfrogs) [inline src/chainparams.cpp:312]: 2021

File: src/chainparams.cpp [hunk 3]
```diff
@@ -302,6 +307,11 @@ class CTestNetParams : public CChainParams {
...
+        consensus.vDeployments[Consensus::DEPLOYMENT_ENFORCE_VALUE].nOverrideRuleChangeActivationThreshold = 1209; // Approx 60% of 2016
```
Comment (blondfrogs) [inline src/chainparams.cpp:313]: Update to same value as mainnet 1411 - 70%

File: src/rpc/blockchain.cpp [hunk 4]
```diff
@@ -1832,6 +1902,7 @@ static const CRPCCommand commands[] =
     { "blockchain",         "getbestblockhash",       &getbestblockhash,       {} },
     { "blockchain",         "getblockcount",          &getblockcount,          {} },
     { "blockchain",         "getblock",               &getblock,               {"blockhash","verbosity|verbose"} },
+    { "blockchain",         "decodeblock",            &decodeblock,               {"blockhex",} },
```
Comment (blondfrogs) [inline src/rpc/blockchain.cpp:1905]: extra comma

File: src/rpc/blockchain.cpp [hunk 5]
```diff
@@ -990,6 +1020,45 @@ UniValue getblock(const JSONRPCRequest& request)
     return blockToJSON(block, pblockindex, verbosity >= 2);
 }
 
+UniValue decodeblock(const JSONRPCRequest& request)
```
Comment (akshaynexus) [inline src/rpc/blockchain.cpp:1023]: Why was this added,doesnt getblock "blockhashhere" achive the same end result?

Comment (akshaynexus) [reply src/rpc/blockchain.cpp:1023]: Nvm,looks like this takes the rawblock hex and decodes.
